src/tiny_slam/tiny_slam.cpp

```diff
@@ -18,24 +26,49 @@
 #include "tiny_world.h"
 #include "tiny_grid_cells.h"
 
+/*!
+ * \brief Derived class from GridScanMatcherObserver to publish robot pose
+ *
+ * This class provides to functions to publish robot pose in ros topic
+ */
 class PoseScanMatcherObserver : public GridScanMatcherObserver {
 public:
+
+  /*!
+   * Publishes the robot pose with child frame sm_curr_pose
```

Comment:
Please, describe _what_ the method does not _how_. In this case: "Publishes the robot pose that is tested by the scan matcher at the moment"